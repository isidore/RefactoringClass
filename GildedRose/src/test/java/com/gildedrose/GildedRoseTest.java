package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.DiffMergeReporter;
import org.junit.Test;

@UseReporter(DiffMergeReporter.class)
public class GildedRoseTest {

	@Test
	public void foo() throws Exception {
		String[] names = { "foo" };
		CombinationApprovals.verifyAllCombinations(this::getItem, names);
	}

	public Item getItem(String name) {
		Item[] items = new Item[] { new Item(name, 0, 0) };
		GildedRose app = new GildedRose(items);
		app.updateQuality();
		Item item = app.items[0];
		return item;
	}

}
