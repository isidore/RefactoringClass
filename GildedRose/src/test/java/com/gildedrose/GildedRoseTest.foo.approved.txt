[foo] => Item [name=foo, sellIn=-1, quality=0] 
